control-plane-operator/controllers/hostedcontrolplane/kas/params.go

```diff
@@ -294,7 +294,7 @@ func NewKubeAPIServerParams(ctx context.Context, hcp *hyperv1.HostedControlPlane
 	params.OwnerRef = config.OwnerRefFrom(hcp)
 
 	params.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
-	params.DeploymentConfig.SetDefaults(hcp, kasLabels(), nil)
+	params.DeploymentConfig.SetRequestServingDefaults(hcp, kasLabels(), nil)
```

Comment:
After this effort and the one for https://github.com/openshift/hypershift/pull/2717 where we are introducing yet another contractual label, we need to prio refactoring the package to provide a simpler and less error prone library that enforces all opinions programatically and that's the only way to add things to componenents.